File: Problems/0509.Fibonacci-Number/FibonacciNumber.go
package leetcode

// Runtime: 0 ms, faster than 100.00% of Go online submissions for Fibonacci Number.
func fib(N int) int {
	if N == 1 || N == 2 {
		return 1
	}
	prev := 1
	curr := 1
	count := 0
	for i := 2; i < N; i++ {
		count = prev + curr
		prev = curr
		curr = count
	}
	return count
}

File: leetcode/0509.Fibonacci-Number/FibonacciNumber.py
class Solution:

    # Runtime: 24 ms, faster than 94.52% of Python3 online submissions for Fibonacci Number.
    def fib(self, N: int) -> int:
        if N in (1, 2): return 1
        curr = prev = 1
        count = 0
        for _ in range(2, N):
            count = curr + prev
            prev = curr
            curr = count
        
        return count